lib/src/test/java/vmath/Mat3Test.java

```diff
@@ -51,6 +51,32 @@ void givenNullArray_whenMat3Created_thenThrowsException() {
         assertThrows(IllegalArgumentException.class, () -> new Mat3(a));
     }
 
+    @Test
+    public void givenNaNValues_whenConstructorCalled_thenThrowArithmeticException() {
+        assertThrows(ArithmeticException.class, () -> {
+            new Mat3(1.0f, 2.0f, Float.NaN,
+                    4.0f, 5.0f, 6.0f,
+                    7.0f, 8.0f, 9.0f);
+        });
+    }
+
+    @Test
+    public void givenValidValues_whenConstructorCalled_thenNoExceptionThrown() {
```

Comment:
1) Test Method Renaming: Please rename the test method to givenNaNValues_whenMat3Created_thenThrowsException. This title more accurately reflects the test's purpose and follows our standard naming conventions.
2) First Assert Condition: For the first assertion, ensure that all elements of the Mat3 matrix are set to Float.NaN. 
3) Last Assert Condition: In the last assertion, only the final element of the Mat3 matrix should be Float.NaN. 

Additionally, check the existing Mat4Test to ensure your changes align with the existing code.